webdriver/tests/interaction/element_click.py

```
@@ -0,0 +1,142 @@
+    # 14.1 Step 4
+    session.url = inline("<input type=text value=Federer><div style= \"height: 200vh; width: 5000vh\">")
+
+    # Scroll to the bottom right of the page
+    session.execute_script("window.scrollTo(document.body.scrollWidth, document.body.scrollHeight);")
+    element = session.find.css("input", all=False)
+    # Clear and scroll back to the top of the page
+    response = click(session, element)
+    assert_success(response)
+
+    # Check if element cleared is scrolled into view
+    rect = session.execute_script("return document.getElementsByTagName(\"input\")[0].getBoundingClientRect()")
+
+    pageDict = {}
+
+    pageDict["innerHeight"] = session.execute_script("return window.innerHeight")
+    pageDict["innerWidth"] = session.execute_script("return window.innerWidth")
+    pageDict["pageXOffset"] = session.execute_script("return window.pageXOffset")
+    pageDict["pageYOffset"] = session.execute_script("return window.pageYOffset")
+
+    assert rect["top"] < (pageDict["innerHeight"] + pageDict["pageYOffset"]) and \
+           rect["left"] < (pageDict["innerWidth"] + pageDict["pageXOffset"]) and \
+           (rect["top"] + element.rect["height"]) > pageDict["pageYOffset"] and \
+           (rect["left"] + element.rect["width"]) > pageDict["pageXOffset"]
+
+# TODO
+# Any suggestions on implementation?
+# def test_element_not_interactable(session):
+#     # 14.1 Step 5
+#     assert_error(response, "element not interactable")
+
+def test_element_click_intercepted(session):
+    # 14.1 Step 6
+    session.url = inline("""<input type=button value=Roger style=\"position: absolute; left: 10px; top: 10px\">
+                            <div style=\"position: absolute; height: 100px; width: 100px; background: rgba(255,0,0,.5); left: 10px; top: 5px\"></div>""")
+
+    element = session.find.css("input", all=False)
+    response = click(session, element)
+    assert_error(response, "element not interactable")
+
+
```

Please can you have a test where we have `overflow:hidden` and we need to scroll to the item inside the overflow.

---

Just to clarify, you want a test where an element will be hidden when there is overflow of data. With overflow:hidden, we wouldn't be able to scroll and click on the element and it should throw an error. Is that right?

---

we should be able to scroll to those elements that in an `overflow:hidden`

---

@alchiem Did you address this?

---

Hi @andreastt, I added the test for overflow:hidden and it throws a element not interactable error. @AutomatedTester said that we should be able to scroll to the elements in an overflow:hidden and have a successful click? There is no scroll in overflow:hidden though right?

---

Yes, as I understand it that is supposed to return an element not interactable error because you can’t scroll an element with `overflow: hidden`.